### ISSUE TITLE
Code Refactoring

### ISSUE THREAD

**Issue by yuristsepaniuk** [peterbabukh/learnwords3#2]
* Model ends with Model.js
* Collection ends with Collection.js
* Change Router to ApplicationRouter.js
* remove Words.js, replace it with local json file (client side)
* why you use charset = "utf-8"; on top of each file, http meta is enough, if not, please comment below.
* lets change each file as follows:
```javascript
define(function(require) {
    'use strict';
    require('backbone.localStorage');

    var Backbone = require('backbone');
    var Proxy = require('app/Proxy');
    var i18n = require('i18n!../../js/nls/ru');
    var FieldsetView = 'FieldsetView';
    
    return FieldsetView;
});
```
* don't use trailing comma as in the end of Words.js
* helpers methods are not under AMD, please call define in each javascript file. So formToJson depends on jquery, not a problem, just call require('jquery') within formToJson. Now It works fine, because you mentioned it in requireConfig, please remove formToJson from there. Also do the same for jquery.some. Btw shim is good place for i18n.js. Well done!
* don't use dot in file names, change jquery.some to smth like jquery_some.js, better eliminate jquery mentioning. 
* please create ApplicationController.js with initialize method, you must config $.fn from there before starting Backbone History. Once everything is done, start backbone. I mean that starting history from main.js is not good. Please keep just requireConfig in main.js and call ApplicationController -> initialize.
* create more comments in the code, as much as you can. Try to think why you need to comment exactly this line of code. Let me show you what is clear and what is not with or without comments:
it is clear that you don't want to fetch collection from local store again
```javascript
if ( !localStorage.preventLoad ) {
			wordsCollection = new WordsCollection( Words );
			wordsCollection.each(function(model) {
                model.save({
                    id: model.get("enWord") + model.get("grade")
                });
            });
		}
```
it is not clear why we need to create PreventLoad model to check presence. You can check it using localStorage['words']
```javascript
preventLoad = new PreventLoad({
			id: 'preventLoad'
		});
		preventLoad.save();
```

It is not clear why we pass wordsCollection to Router.js. If I needed it so better record it to window.LW.cache.words = wordsCollection; So that Router is not dependent on collection. 
```javascript
router = new Router( wordsCollection );
```

It is not clear why we need Proxy.js Model, please comment.
* make router free from collection array, router should handle just parameters passing to view, view is responsible for model fetching. Also don't pass router to each view. Router should just pass id of smth to each view, but not heavy metal :+1: 
* good to know that you figure out i18n for yourself. But later think about hardcoded messages in the code. I will create separate ticket for localization.

**Comment by peterbabukh** [comment]
- After deep study of this issue I came to a conclusion that localStorage works in such a way that I have  to keep reference to the same collection object after initialization, because otherwise different collection objects will be fetched and when i route from one view to another , they will not fetch the updated collection correctly. So I decided to leave it as it is passing the collection within the router from one view to another to keep the reference.  Thus the only way i see is to pass the reference explicitly. Well, there is another way - get and parse the stored models by means of localStorage methods instead of RESTful backbone, but it will take too many useless efforts to refactor.As the localStorage is for local usages only and the app is about to be totally refactored to be served on the server, I decided to leave it hardcoded.

- Also I will launch Json file instead of words.js on my way to server implementation of the app.

- From now on I will keep to reasonable commenting according to your tips. Later or on demand I will add comments to the existing code.

- Thus, basically it is the restyling that was done. Hope it will be enough to merge and continue further work over the new code style. As far as i see it , the implementation of the server side will lead to further changes in the architecture and logic. But it's gonna be another story.
